### ISSUE TITLE
Custom import script gets cleared on project exit

### ISSUE THREAD

**Issue by ctrlaltmilk** [godotengine/godot#93729]
### Tested versions

- Reproducible in 4.3-beta2
- Not reproducible in 4.2.1-stable

### System information

Godot v4.3.beta2 - KDE neon 6.0 22.04 - Wayland - GLES3 (Compatibility) - Mesa Intel(R) Graphics (ADL GT2) - 12th Gen Intel(R) Core(TM) i5-1240P (16 Threads)

### Issue description

Importing a resource with a custom import script works at first, but when attempting to exit the project, it reverts to its default importer. This breaks a lot of use cases for import scripts, as they now have to be manually attended to upon every project load. Copying the file while Godot is open, then pasting after the project is closed works. However, this also gets reverted upon the next project exit.

### Steps to reproduce

- Open reproduction project.
- Verify that `test.csv` is correctly imported using the test import script.
- Close project.
- Reopen project.
- Verify that `test.csv` has reverted to its default importer.

### Minimal reproduction project (MRP)

[import-script-mrp.zip](https://github.com/user-attachments/files/16039064/import-script-mrp.zip)

**Comment by RadiantUwU** [comment]
Confirmed, the importer is changed once the project is closed.

**Comment by RadiantUwU** [comment]
Found the issue
This lies within the MRP, using `_exit_tree` makes it remove the importer before godot actually finished shutting down, instead of using `_disable_plugin` which is what it's actually meant for it.
```gdscript
@tool
extends EditorPlugin

const ImporterPlugin = preload("importer_plugin.gd")

var import_plugin

func _enter_tree() -> void:
	import_plugin = ImporterPlugin.new()
	add_import_plugin(import_plugin)

func _disable_plugin() -> void:
	remove_import_plugin(import_plugin)
	import_plugin = null
```
This is how the correct version would look, where the bug is unable to be reproduced anymore.
This is not a bug, this is just an issue on how plugins were used for this.

**Comment by ctrlaltmilk** [comment]
If that is the case, I think this needs to be mentioned somewhere - considering many, many resource import plugins exist, and `_exit_tree` has been the definitive place to remove import plugins, there's a significant chance that any plugin not made specifically for 4.3 will not work correctly.

**Comment by Hilderin** [comment]
I tested the MRP with a build from master and I'm not able to reproduce the problem.
I think it was fixed by: #93518
Now that a `if (!EditorNode::get_singleton()->is_exiting())` is added, the importer is not reverted back to the default one when the plugin is removed in `_exit_tree`, keeping the old behavior for plugins.

**Comment by akien-mga** [comment]
Thanks, closing as fixed by #93518.
